gems/smithy/spec/spec_helper.rb

```diff
@@ -1,77 +1,47 @@
-    #  `:schema`, `:client`, or `:server`.
-    # @param [Hash] options Additional options to pass to the generator.
-    # @option options [String] :fixture The name of the fixture to load.
-    # @return [String] The path to the directory where the generated code was
-    #  written to.
-    def generate(modules, type, options = {})
-      model = load_model(modules, options)
-      plan = create_plan(modules, model, type, options)
-      sdk_dir = plan.destination_root
-      Smithy.generate(plan)
-
-      $LOAD_PATH << ("#{sdk_dir}/lib")
+    # (See ClientHelper#generate)
+    def generate_gem(type, options = {})
+      plan = ClientHelper.generate(type, options)
+      $LOAD_PATH << "#{plan.destination_root}/lib"
       require plan.gem_name
-
-      RbsSpyTest.setup(modules, sdk_dir) if ENV.fetch('SMITHY_RUBY_RBS_TEST', false)
-      sdk_dir
-    rescue StandardError => e
-      cleanup(modules, sdk_dir) if sdk_dir
+      RbsSpyTest.setup(plan.module_name, plan.destination_root) if ENV['SMITHY_RUBY_RBS_TEST']
+      plan
+    rescue LoadError => e
+      puts "Error loading gem: #{plan.gem_name}"
       raise e
     end
 
-    # @param [Array<String>] module_names A list of module names from the
-    #  generated code to clean up.
-    # @param [String] tmpdir The path to the tmp directory where the
-    #  generated code was written to.
-    def cleanup(module_names, tmpdir)
-      return unless tmpdir
-
-      if ENV['SMITHY_RUBY_KEEP_GENERATED_SOURCE']
-        puts "Leaving generated service in: #{tmpdir}"
-      else
-        FileUtils.rm_rf(tmpdir)
-      end
-      $LOAD_PATH.delete("#{tmpdir}/lib")
-      module_names.reverse.each_cons(2) do |child, parent|
-        Object.const_get(parent).send(:remove_const, child)
-      end
-      Object.send(:remove_const, module_names.first)
+    # (See ClientHelper#source)
+    def generate_from_source_code(type, options = {})
+      module_name, source = ClientHelper.source(type, options)
+      Object.module_eval(source)
+      Object.const_get(module_name)
+      module_name
+    rescue LoadError => e
+      puts "Error evaluating source:\n#{source}"
+      raise e
     end
 
-    private
-
-    def load_model(modules, options)
-      fixture = options[:fixture] || modules.map(&:underscore).join('/')
-      model_dir = File.join(File.dirname(__FILE__), 'fixtures', fixture)
-      JSON.load_file(File.join(model_dir, 'model.json'))
+    def cleanup_gem(plan)
+      ClientHelper.cleanup_gem(plan.module_name, plan.destination_root)
     end
 
-    def create_plan(modules, model, type, options)
-      plan_options = {
-        module_name: modules.join('::'),
-        gem_version: options.fetch(:gem_version, '0.1.0'),
-        destination_root: options.fetch(:destination_root, Dir.mktmpdir),
-        quiet: ENV.fetch('SMITHY_RUBY_QUIET', 'true') == 'true'
-      }
-      Smithy::Plan.new(model, type, plan_options)
+    def cleanup_eval_source(module_name)
```

Comment:
this reads a little weirdly to me - isn't this cleaning up modules? could we just call it "cleanup_modules" or "remove_generated_modules"